src/thread_pool.rs

```diff
@@ -112,11 +116,14 @@ impl Registry {
     pub unsafe fn inject(&self, injected_jobs: &[*mut Job<LockLatch>]) {
         log!(InjectJobs { count: injected_jobs.len() });
         let mut state = self.state.lock().unwrap();
+        if state.terminate {
+            panic!("attempted to inject a job after thread pool shut down");
```

Comment:
This error message will be seen by end-users, right? That is, if a panic occurs in one place, it would get propagated to later jobs?

Perhaps we should make it a bit more readable, something like: 

"rayon thread pool is contaminated by a previous panic; recovery is currently unimplemented, see <https://github.com/nikomatsakis/rayon/issues/10> for details"